### PR TITLE
add android-setup command

### PR DESCRIPTION
**Pull request by jjaffeux**
This pull request adds a new command : android-setup. This command allows users to install android SDK and NDK, and also keep it updated.

Usage : `motion android-setup`

Todo :
- [x] Change DefaultDirectory, while reviewing this PR it's currently set to be installed in ~/rubymotion-android-test
- [x] Simplify PACKAGES_LIST, it currently installs all system-images and it's probably not needed
- [ ] Improve handling of existing android SDK/NDK installation

### CODE REVIEW

File: lib/motion/command/android_setup.rb [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        FileUtils.mv(File.join(@tmp_directory, "android-ndk-#{@ndk_version}"), @ndk_directory)
+      else
+        $stderr.puts("Installed NDK is up to date.")
+      end
+    end
+
+    def current_ndk_version
+      file = File.open(File.join(@ndk_directory, 'RELEASE.TXT'), "rb")
+      version = file.read.strip.split(' ')[0]
+      file.close
```
Comment (lrz) [inline lib/motion/command/android_setup.rb]: Could be replaced with:

`version = File.read(File.join(@ndk_directory, 'RELEASE.TXT'). strip.split(' ')[0]`

File: lib/motion/command/android_setup.rb [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+      version
+    end
+
+    def setup_sdk
+      $stderr.puts("-- Setup SDK...")
+      if system("which #{android_executable} > /dev/null")
```
Comment (lrz) [inline lib/motion/command/android_setup.rb]: Better to do

`if File.exist?(android_executable)`

File: lib/motion/command/android_setup.rb [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        $stderr.puts("Found installed SDK, trying to update...")
+      else
+        $stderr.puts("Installing SDK : version #{@sdk_version}")
+        sdk_url = File.join(DL_GOOGLE, "android-sdk_r#{@sdk_version}-macosx.zip")
+        zipped_sdk = File.join(@tmp_directory, "sdk.zip")
+        download(sdk_url, zipped_sdk)
+        extracted_sdk = File.join(@tmp_directory, 'sdk')
+        system("/usr/bin/unzip -q -a #{zipped_sdk} -d #{extracted_sdk}")
```
Comment (lrz) [inline lib/motion/command/android_setup.rb]: Better to escape paths:

`... \"#{zipped_sdk}\" -d \"#{extracted_sdk}\"...`

File: lib/motion/command/android_setup.rb [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    def create_directory_structure
+      $stderr.puts("-- Create directory structure...")
+      FileUtils.mkdir_p(@tmp_directory)
+      $stderr.puts(@directory)
+      $stderr.puts(@tmp_directory)
+    end
+
+    def check_java
+      $stderr.puts("-- Checking Java is installed...")
+      unless system("which /usr/bin/javac > /dev/null")
```
Comment (lrz) [inline lib/motion/command/android_setup.rb]: Better to do this:

`unless File.exist?('/usr/bin/java')`

File: lib/motion/command/android_setup.rb [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+      resp = `/usr/bin/curl --connect-timeout 60 #{cmd}`
+      if $?.exitstatus != 0
+        die("Error when connecting to the server. Check your Internet connection and try again.")
+      end
+      resp
+    end
+
+    def download(url, dest)
+      if system("which axel > /dev/null")
+        unless system("axel -n 10 -a -o '#{dest}' '#{url}'")
```
Comment (lrz) [inline lib/motion/command/android_setup.rb]: Here we should be using the path to axel in the $PATH.

Probably something like this:

```
axel_path = `/usr/bin/which axel`
if $?.success?
  unless system("#{axel_path.strip} -n ...")
else
  curl(...)
end
```

**Comment by MarkVillacampa** [comment]
This has been integrated.

**Pull request closed without merging**